File: Products of Array Discluding Self/ts/productsOfArrayDiscludingSelf.js
function productExceptSelf(nums) {
    var result = [];
    var prefix = 1;
    var sufix = 1;
    for (var i = 0; i < nums.length; i++) {
        result[i] = prefix;
        prefix *= nums[i];
    }
    for (var i = nums.length - 2; i >= 0; i--) {
        sufix *= nums[i + 1];
        result[i] *= sufix;
    }
    return result;
}
var nums = [1, 2, 4, 6];
console.log(productExceptSelf(nums));

File: Products of Array Discluding Self/python/productsOfArrayDiscludingSelf.py
from typing import List


def productExceptSelf(nums: List[int]) -> List[int]:
    res = [1] * len(nums)

    for i in range(1, len(nums)):
        res[i] = res[i - 1] * nums[i - 1]

    postfix = 1
    for i in range(len(nums) - 1, -1, -1):
        res[i] *= postfix
        postfix *= nums[i]

    return res

nums = [1,2,4,6]
print(productExceptSelf(nums))